packages/bigtable/src/instance.js

```diff
@@ -0,0 +1,665 @@
+     *     instance as it appears in UIs. Can be changed at any time, but should
+     *     be kept globally unique to avoid confusion.
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * instance.setMetadata({
+     *   displayName: 'updated-name'
+     * }, function(err, apiResponse) {});
+     */
+    setMetadata: {
+      protoOpts: {
+        service: 'BigtableInstanceAdmin',
+        method: 'updateInstance'
+      },
+      reqOpts: {
+        name: id
+      }
+    }
+  };
+
+  var config = {
+    parent: bigtable,
+    id: id,
+    methods: methods,
+    createMethod: function(_, options, callback) {
+      bigtable.createInstance(name, options, callback);
+    }
+  };
+
+  common.GrpcServiceObject.call(this, config);
+}
+
+nodeutil.inherits(Instance, common.GrpcServiceObject);
+
+/**
+ * Formats the full table name into a user friendly version.
+ *
+ * @private
+ *
+ * @param {string} name - The formatted Table name.
+ * @return {string}
+ *
+ * @example
+ * Instance.formatTableName_('projects/p/zones/z/clusters/c/tables/my-table');
+ * // => 'my-table'
+ */
+Instance.formatTableName_ = function(name) {
+  if (name.indexOf('/') === -1) {
+    return name;
+  }
+
+  var parts = name.split('/');
+  return parts[parts.length - 1];
+};
+
+/**
+ * Creates a cluster.
+ *
+ * @param {string} name - The name to be used when referring to the new
+ *     cluster within its instance.
+ * @param {object=} options - Cluster creation options.
+ * @param {string} options.location - The location where this cluster's nodes
+ *     and storage reside. For best performance clients should be located as
+ *     as close as possible to this cluster. Currently only zones are
+ *     supported.
+ * @param {number} options.nodes - The number of nodes allocated to this
+ *     cluster. More nodes enable higher throughput and more consistent
+ *     performance.
+ * @param {string} options.storage - The type of storage used by this cluster
+ *     to serve its parent instance's tables. Options are 'hdd' or 'ssd'.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:bigtable/cluster} callback.cluster - The newly created
+ *     cluster.
+ * @param {Operation} callback.operation - An operation object that can be used
+ *     to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var callback = function(err, cluster, operation, apiResponse) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   operation
+ *     .on('error', console.log)
+ *     .on('complete', function() {
+ *       // The cluster has successfully been created.
```

Comment:
To match the convention in our other APIs, this would be "The cluster was created successfully."